utils/dataloaders.py

```diff
@@ -494,15 +509,22 @@ def __init__(self,
                     # f = list(p.rglob('*.*'))  # pathlib
                 elif p.is_file():  # file
                     with open(p) as t:
+                        # Read contents of txt file
                         t = t.read().strip().splitlines()
                         parent = str(p.parent) + os.sep
-                        f += [x.replace('./', parent, 1) if x.startswith('./') else x for x in t]  # to global path
+                        f += [x.replace('./', parent, 1) if x.startswith('./') else input_dir + x for x in t]  # to global path
                         # f += [p.parent / x.lstrip(os.sep) for x in t]  # to global path (pathlib)
+
                 else:
                     raise FileNotFoundError(f'{prefix}{p} does not exist')
-            self.im_files = sorted(x.replace('/', os.sep) for x in f if x.split('.')[-1].lower() in IMG_FORMATS)
-            # self.img_files = sorted([x for x in f if x.suffix[1:].lower() in IMG_FORMATS])  # pathlib
-            assert self.im_files, f'{prefix}No images found'
+            images_to_process = sorted(x.replace('/', os.sep) for x in f if x.split('.')[-1].lower() in IMG_FORMATS)
+
+            # Create a list of images to be processed that are not in the list of processed images
+            self.im_files = [image for image in images_to_process if extract_folder_and_filename(image) not in processed_images]
```

Comment:
Regarding the images to be processed, shall we also not check their status in the database?

Reply:
Images_to_process are all images that the application found in the database.
processed_images are all the images in the database that have the label "inprogress" or "processed"

I will add this info also in the dataloader.py as comments